src/Evolution/Systems/ScalarWave/BoundaryConditions/ConstraintPreservingSphericalRadiation.hpp

```diff
@@ -0,0 +1,264 @@
+ * \f}
+ *
+ * The constraints are defined as:
+ *
+ * \f{align*}{
+ *  \mathcal{C}_i&=\partial_i\Psi - \Phi_i=0, \\
+ *  \mathcal{C}_{ij}&=\partial_{[i}\Phi_{j]}=0
+ * \f}
+ *
+ * Radiation boundary conditions impose a condition on \f$\Pi\f$ or its time
+ * derivative. We denote the boundary condition value of the time derivative of
+ * \f$\Pi\f$ by \f$\partial_t\Pi^{\mathrm{BC}}\f$. With this, we can impose
+ * boundary conditions on the time derivatives of the evolved variables follows:
+ *
+ * \f{align*}{
+ * \partial_{t} \Psi&\to\partial_{t}\Psi +
+ *                    \lambda_\Psi n^i \mathcal{C}_i, \\
+ * \partial_{t}\Pi&\to\partial_{t}\Pi-\left(\partial_t\Pi
+ *                  - \partial_t\Pi^{\mathrm{BC}}\right)
+ *                  +\gamma_2\lambda_\Psi n^i \mathcal{C}_i
+ *                  =\partial_t\Pi^{\mathrm{BC}}
+ *                  +\gamma_2\lambda_\Psi n^i \mathcal{C}_i, \\
+ * \partial_{t}\Phi_i&\to\partial_{t}\Phi_i+
+ *                     \lambda_0n^jP^k{}_i\mathcal{C}_{jk}
+ *   = \partial_{t}\Phi_i+ \lambda_0n^j \mathcal{C}_{ji}.
+ * \f}
+ *
+ * Below we assume the normal vector \f$n^i\f$ is the radial unit normal vector.
+ * That is, we assume the outer boundary is spherical. A Sommerfeld radiation
+ * condition is given by
+ *
+ * \f{align*}{
+ *  \partial_t\psi=n^i\Phi_i
+ * \f}
+ *
+ * Or, assuming that \f$\partial_tn^i=0\f$ (or is very small),
+ *
+ * \f{align*}{
+ *  \partial_t\Pi^{\mathrm{BC}}=n^i\partial_t\Phi_i
+ * \f}
+ *
+ * The Bayliss-Turkel boundary conditions are given by:
+ *
+ * \f{align*}{
+ *  \left(\partial_t + \partial_r + r^{-1}\right)^m\Psi=0
+ * \f}
+ *
+ * The first-order form is
+ *
+ * \f{align*}{
+ *  \partial_t\Pi^{\mathrm{BC}}=n^i\partial_t\Phi_i + \frac{1}{r}\partial_t\Psi,
+ * \f}
+ *
+ * assuming \f$\partial_t n^i=0\f$ and \f$\partial_t r=0\f$.
+ *
+ * The second-order boundary condition is given by,
+ *
+ * \f{align*}{
+ *  \partial_t\Pi^{\mathrm{BC}}
+ *   &=\left(\partial_t\partial_r + \partial_r\partial_t +
+ *     \partial_r^2+\frac{4}{r}\partial_t
+ *     +\frac{4}{r}\partial_r + \frac{2}{r^2}\right)\Psi \notag \\
+ *     &=n^i(\partial_t\Phi_i-\partial_i\Pi) + n^i n^j\partial_i\Phi_j -
+ *     \frac{4}{r}\Pi+\frac{4}{r}n^i\Phi_i + \frac{2}{r^2}\Psi,
+ * \f}
+ *
+ * assuming \f$\partial_t n^i=0\f$ and \f$\partial_t r=0\f$.
+ *
+ * The moving mesh can be accounted for by using
+ *
+ * \f{align*}{
+ * \partial_t r = \frac{1}{r}x^i\delta_{ij}\partial_t x^j
+ * \f}
+ *
+ * \note It is not clear if \f$\partial_t\Phi_i\f$ should be replaced by
+ * \f$-\partial_i\Pi\f$, which is the evolution equation but without the
+ * constraint.
+ *
+ * \warning The boundary conditions are implemented assuming the outer boundary
+ * is spherical. It might be possible to generalize the condition to
+ * non-spherical boundaries by using \f$x^i/r\f$ instead of \f$n^i\f$, but this
+ * hasn't been tested.
+ */
+template <size_t Dim>
+class ConstraintPreservingSphericalRadiation final
```

Comment:
Would it be useful/feasible to merge this into `SphericalRadiation` with an option `ConstraintPreserving: True/False`?

Reply:
Nope, not at all. `SphericalRadiation` is applied via ghost cells while this is applied via a Bjorhus method. Part of the reason for having both is to be able to verify that Bjorhus works and that the spherical radiation conditions work at all. They are, however, different boundary conditions. `SphericalRadiation` imposes a weak radiation condition while `ConstraintPreservingSphericalRadiation` imposes it strongly on the time derivatives.